samples/nrf9160/modem_shell/src/main.c

```diff
@@ -208,6 +208,16 @@ static void button_handler(uint32_t button_states, uint32_t has_changed)
 }
 #endif
 
+#if !defined(CONFIG_LWM2M_CARRIER)
+static void system_reboot(void)
+{
+	/* Set logger subsystem to panic mode to flush pending logs before the reboot. */
+	LOG_PANIC();
+
+	sys_reboot(SYS_REBOOT_WARM);
```

Comment:
There's still the fact that for me `log_panic()` did not actually flush as intended. @tokangas you saw the same issue right?

Reply:
@lemrey Yes, currently there's an issue with this. While logger does not currently work as expected, I though it would be good practice to add it anyway.

I looked at the code and as far as I understood logger enters synchronous mode (same as `CONFIG_LOG_MODE_IMMEDIATE`) when `log_panic()` is called.

Reply:
The issue was caused by `CONFIG_LOG_PRINTK` and the shell UART backend being enabled too late to output the traces. I updated the commit to just disable `CONFIG_LOG_PRINTK` to keep `printk()` synchronous.